authors/bruno-skvorc/polkadot/en_US/final.md

```diff
@@ -0,0 +1,186 @@
+
+### Slashing
+
+Reasons for slashing are the same in both systems:
+
+- producing two blocks for the same slot
+- signing two conflicting chains
+- being offline
+
+Additionally, in Polkadot there's the penalty for illegal randomness rolls which can be detected by fishermen. See Randomness section above for details.
+
+In Polkadot any offence not only slashes, but also removes a validator from the validator set in the current era. They have to re-qualify to rejoin.
+
+## Smart Contracts, Execution Environments, Runtimes and SPREE
+
+Finally, let's talk bout the meat of the protocol - the execution layer.
+
+In Ethereum, there will be the concept of Execution Enivronments. These will be special runtimes built by special teams and funded by special interests to run and execute custom code. In other words, if someone wants to build the ability to process bitcoin-like transactions into Ethereum, they need to fund a UTXO-execution-environment and make sure all clients add it to their code as well (likely helping with funding those efforts, or contributing outright). Once done, the execution environment becomes available to everyone - it's considered a network upgrade.
+
+The first and default execution environment will be eWASM, a neutered version of WASM flavored for Ethereum. The second one may very well be an EVM EE which would allow Ethereum 1 to continue living and breathing inside of Ethereum 2, perhaps on a dedicated shard. Developing custom EEs will be expensive and hard, so we don't expect to see many of them.
+
+eWASM will support smart contracts through languages that can compile to WASM like [Nimplay](https://github.com/status-im/nimplay/). EVM EE will support smart contracts just like Ethereum does today.
+
+In Polkadot, the relay chain does not have smart contract support. Given that there are no parachains yet, there is effectively nowhere to go if you want to develop a smart contract of any kind on Polkadot. However, Substrate - the framework on which the relay chain is built, and the framework which others will likely use to deploy parachains, has support for modules - blobs of plug-and-play code used to enhance your chain's runtime with new functionality. One such module is the Smart Contracts module which, when plugged into a parachain, can support smart contract execution. There are two caveats to such a model:
+
+1. When you want to just fire up a smart contract, you have to do it on a specific smart-contract parachain (though nothing prevents you from deploying the same thing on all smart contract parachains). Picking between them might be difficult - they can be custom-tweaked for specific use cases (privacy vs speed, for example).
+2. That smart contract chain has to communicate with other chains to be of any use, otherwise you're working in a silo.
```

Comment:
Why can't it be of use if you're in a silo?  The current Ethereum chain is basically a silo but I can run all sorts of interesting stuff on it.

Reply:
True. Reworded.